torch/csrc/jit/codegen/cuda/test/test_gpu.cpp

```diff
@@ -24079,6 +24079,72 @@ TEST_F(NVFuserTest, FusionIssue1785Repro_CUDA) {
   testValidate(&fusion, cg_outputs, {in1, in2}, {tv_ref}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionSkipReplay_CUDA) {
+  struct TransformPropagatorWithCheck : public TransformPropagator {
```

Comment:
Do we want to make this a global class in our test cases, and replace all `TransformPropagator`s with `TransformPropagatorWithCheck`? If we want to do this, it will be a separate PR.

Reply:
I don't see the harm in doing so, if you think it'll help us catch issues in the future, sure!